packages/core/src/lib/relay/index.ts

```diff
@@ -208,6 +183,12 @@ class Relay extends GossipSub implements IRelay {
   getMeshPeers(topic?: TopicStr): PeerIdStr[] {
     return super.getMeshPeers(topic ?? this.pubSubTopic);
   }
+
+  public getObservers<T extends IDecodedMessage>(
```

Comment:
I don't think it is any better than just exposing `observers`...

@fryorcraken

Reply:
By `dump` method, I would mean a method that returns a `string` rendered version of the observers for logging purposes.

Not sure what are common practice in JavaScript. For example, in Rust, one would implement the `fmt::Debug` trait.

Reply:
I have an idea. Will write once back on the laptop.

Reply:
The idea is to inspire ourselves from the new filter protocol https://github.com/vacp2p/rfc/pull/562/files and see if we can provide an interface on active subscription that could then be added to `IReceiver`.

Look at the filter request, it has 1 optional pubsub and an array of content topics.
There is a also a new "ping" request to check that the remote has active subscriptions for the local node.

Hence, we could have an interface such as:

```ts
interface ActiveSubscriptions {
    pubsubTopic: string;
    contentTopics: string[];
}

interface IReceiver {
    // ... subscribe
    getActiveSubscriptions: () => Promise<ActiveSubscriptions | undefined>
}
```

- In the case of relay, the `Promise` would resolve instantly.
- In the case of the new filter protocol, we can do a `ping` to the remote server and return `undefined` if remote server returns an error or returned locally saved subscriptions if `200 OK` is returned.

Reply:
Looks really neat. 👍  I will update this PR to close original issue and update IReceiver PR to correlate with this. 

One thing I what to change is `Promise<ActiveSubscriptions | undefined>` - I believe if operation is sync it shouldn't be overcomplicated with Promise.

Reply:
Agree, let's handle it when time comes